### ISSUE TITLE
Range error when dragging the indicator too far in right direction

### ISSUE THREAD

**Issue by hirschmann** [SplashByte/animated_toggle_switch#20]
On desktop, when you drag the indicator with the mouse and release it outside of the right edge of the widget, a RangeError is thrown. The indicator snaps into the right place but onChanged() is not executed and the widget doesn't react properly anymore.

**Comment by MaddinMade** [comment]
I'm sorry for the late reply. I can't reproduce this error. Is this error thrown only with Flutter Desktop? Actually, no value should be passed at this point in the code that causes a RangeError.

**Comment by hirschmann** [comment]
It only happens on Flutter Desktop.
I have only tried the rolling switch yet, so I can't tell anything about the other switches.

I can reproduce the error with this example:
```dart
// ...
Padding(
  padding: EdgeInsets.symmetric(vertical: kIsDesktop ? 9.0 : 15.0, horizontal: 0),
  child: AspectRatio(
    aspectRatio: 2,
    child: AnimatedToggleSwitch<ThemeMode>.rolling(
      current: _themeMode,
      iconRadius: 7,
      selectedIconRadius: 7,
      animationDuration: Duration(milliseconds: 400),
      borderWidth: 1,
      minTouchTargetSize: 100,
      borderColor: Color.fromARGB(24, 175, 175, 175),
      values: [ThemeMode.light, ThemeMode.dark],
      onChanged: (m) => widget.switchThemeAction(),
      iconBuilder: (ThemeMode m, Size s, bool b) => m == ThemeMode.dark
          ? Icon(Icons.dark_mode, size: s.longestSide, color: Color.fromARGB(255, 171, 182, 196))
          : Icon(Icons.light_mode, size: s.longestSide, color: Color.fromARGB(255, 228, 206, 164)),
    ),
  ),
)
// ...
```

It seems like the issue occurs because I wrapped AnimatedToggleSwitch in an `AspectRatio` widget.

> Flutter 2.10.3 • channel stable • https://github.com/flutter/flutter.git
> Framework • revision 7e9793dee1 (7 weeks ago) • 2022-03-02 11:23:12 -0600
> Engine • revision bd539267b4
> Tools • Dart 2.16.1 • DevTools 2.9.2

**Comment by MaddinMade** [comment]
Thank you for the detailed answer. I would just like to understand the cause of this error. If I can't figure it out, I'll apply your fix, though that will fix the symptom rather than the problem.
I'll have a look tomorrow with Desktop, since I unfortunately don't have the possibility to test it with Desktop at the moment.

**Comment by MaddinMade** [comment]
I have not been able to reproduce the error, but I have found the cause. Unfortunately, this is my error due to my implementation. The widget currently does not support tight BoxConstraints as passed by AspectRatio. You have to control the size via the attributes. I should refactor that. I'm sorry you had problems because of that. So currently the fix is probably to only pass BoxConstraints to the widget that are not tight. Otherwise the widget works very bad.

**Comment by MaddinMade** [comment]
After all, it was only a few lines that I had to change. If you can confirm that the problem is fixed in version 0.5.0, this issue can be closed. Thanks for reporting and the pull request, even if it probably won't be merged.

**Comment by MaddinMade** [comment]
@hirschmann Sorry, I wanted to upload the new version for a while anyway, which is why I did it too fast. If version 0.5.0 didn't fix your problem, then maybe 309ccabb9465ad04d8bf0cfe3ee44aef4efbd2f0. It would be nice if you could let me know if one of the two versions fixes your problem :)

**Comment by hirschmann** [comment]
Indeed, 0.5.0 didn't fix the issue, but https://github.com/SplashByte/animated_toggle_switch/commit/309ccabb9465ad04d8bf0cfe3ee44aef4efbd2f0 does.
Thank you very much for your fast response and the fast fix as well :)

**Comment by MaddinMade** [comment]
@hirschmann No problem :)
I still wouldn't use AspectRatio with this widget, because it forces it to a shape, which doesn't always look the way you want it to as a user.
I wish you a lot of fun with this package.

**Comment by MaddinMade** [comment]
The fix is now available with version 0.5.1 on pub.dev
